airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractPostgresSourceDatatypeTest.java

```diff
@@ -0,0 +1,526 @@
+              .addExpectedValues("a", "abc", "Миші йдуть на південь, не питай чому;", "櫻花分店", "",
+                  null, "\\xF0\\x9F\\x9A\\x80")
+              .build());
+    }
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("varchar")
+            .fullSourceDataType("character varying(10)")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("'{asb123}'", "'{asb12}'")
+            .addExpectedValues("{asb123}", "{asb12}")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("cidr")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("null", "'192.168.100.128/25'", "'192.168/24'", "'192.168.1'",
+                "'128.1'", "'2001:4f8:3:ba::/64'")
+            .addExpectedValues(null, "192.168.100.128/25", "192.168.0.0/24", "192.168.1.0/24",
+                "128.1.0.0/16", "2001:4f8:3:ba::/64")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("circle")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("'(5,7),10'", "'(0,0),0'", "'(-10,-4),10'", "null")
+            .addExpectedValues("<(5.0,7.0),10.0>", "<(0.0,0.0),0.0>", "<(-10.0,-4.0),10.0>", null)
+            .build());
+
+    // Debezium does not handle era indicators (AD nd BC)
+    // https://github.com/airbytehq/airbyte/issues/14590
+    // addDataTypeTestData(
+    // TestDataHolder.builder()
+    // .sourceType("date")
+    // .airbyteType(JsonSchemaType.STRING_DATE)
+    // .addInsertValues("'1999-01-08'", "'1991-02-10 BC'", "null")
+    // .addExpectedValues("1999-01-08", "1990-02-10 BC", null)
+    // .build());
+
+    for (final String type : Set.of("double precision", "float", "float8")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.NUMBER)
+              .addInsertValues("'123'", "'1234567890.1234567'", "null")
+              // Postgres source does not support these special values yet
+              // https://github.com/airbytehq/airbyte/issues/8902
+              // "'-Infinity'", "'Infinity'", "'NaN'", "null")
+              .addExpectedValues("123.0", "1.2345678901234567E9", null)
+              // "-Infinity", "Infinity", "NaN", null)
+              .build());
+    }
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("inet")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("'198.24.10.0/24'", "'198.24.10.0'", "'198.10/8'", "null")
+            .addExpectedValues("198.24.10.0/24", "198.24.10.0", "198.10.0.0/8", null)
+            .build());
+
+    for (final String type : Set.of("integer", "int", "int4")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.NUMBER)
```

Comment:
fyi there are some merge conflicts after https://github.com/airbytehq/airbyte/pull/14362 - ints now generate `JsonSchemaType.INTEGER` (same comment for smallint, etc)